src/app/analytics/analytics.factory.js

```diff
@@ -5,18 +5,54 @@ import find from 'lodash/find'
 import sha3 from 'crypto-js/sha3'
 import merge from 'lodash/merge'
 import isEmpty from 'lodash/isEmpty'
+import moment from 'moment'
 /* global localStorage */
 
+
+function suppressErrors (func) {
+  return function wrapper (...args) {
+    try {
+      return func.apply(this, args)
+    } catch(e) { }
+  }
+}
+// Generate a datalayer product object
+const generateProduct = suppressErrors(function(item, additionalData = {}) {
+  const price = additionalData?.price || item?.price
+  const category = additionalData?.category || item.designationType
+  const name = additionalData?.name || item.displayName || undefined
+  const recurringDate = additionalData.recurringDate
+    ? additionalData.recurringDate.format('MMMM D, YYYY')
+    : item.giftStartDate
+    ? item.giftStartDate.format('MMMM D, YYYY')
+    : undefined
+  const frequencyObj = find(item.frequencies, { name: item.frequency })
+  const variant = frequencyObj?.display || item.frequency
+  const itemVariant = variant ? variant.toLowerCase() : undefined
```

Comment:
Maybe I'm missing something, but I would disagree, if it was inlined it would be hard to read, also we would need to ensure `frequencyObj?.display` is not undefined, as trying to run `.toLowerCase()` on undefined would result in an error.

We would have to write it as
```
{
item_variant: frequencyObj?.display ? frequencyObj?.display.toLowerCase() :  item.frequency ? item.frequency.toLowerCase() : undefined,
}
```

Reply:
I was more thinking of
```
{
item_variant: variant ? variant.toLowerCase() : undefined,
}
```

Reply:
That makes sense